src/main/java/com/coveo/pushapiclient/StreamUpdateRecord.java

```diff
@@ -0,0 +1,48 @@
+package com.coveo.pushapiclient;
+
+import com.google.gson.JsonObject;
+import java.util.Arrays;
+
+public class StreamUpdateRecord extends BatchUpdateRecord {
+
+  private final JsonObject[] partialUpdate;
+
+  public StreamUpdateRecord(
+      JsonObject[] addOrUpdate, JsonObject[] delete, JsonObject[] partialUpdate) {
+    super(addOrUpdate, delete);
+    this.partialUpdate = partialUpdate;
+  }
+
+  public JsonObject[] getPartialUpdate() {
+    return partialUpdate;
+  }
+
+  @Override
+  public String toString() {
+    return "StreamUpdateRecord["
+        + "addOrUpdate="
+        + Arrays.toString(this.getAddOrUpdate())
+        + ", delete="
+        + Arrays.toString(this.getDelete())
+        + ", partialUpdate="
+        + Arrays.toString(partialUpdate)
+        + ']';
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
We need the equals twice because we are also comparing that the object type is the same. If we were to call the super.equals we would be trying to compare a parent with it's child class